core/finality-grandpa/src/until_imported.rs

```diff
@@ -584,22 +585,21 @@ mod tests {
 		// enact all dependencies before importing the message
 		enact_dependencies(&chain_state);
 
-		let (global_tx, global_rx) = futures::sync::mpsc::unbounded();
+		let (global_tx, global_rx) = futures::channel::mpsc::unbounded();
 
 		let until_imported = UntilGlobalMessageBlocksImported::new(
 			import_notifications,
 			TestBlockSyncRequester::default(),
 			block_status,
-			global_rx.map_err(|_| panic!("should never error")),
+			global_rx.map_err(|()| panic!("should never error")),
```

Comment:
This needs a better justification.  If there is no possible error, we should use a void type here.  It is possible to implement a void type without an ICE.